test/e2e/clusterclass_upgrade_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
```

Comment:
We can probably look at arranging our upgrade tests similar to what CAPI does - https://github.com/kubernetes-sigs/cluster-api/blob/main/test/e2e/cluster_upgrade_test.go
WDYT?

Reply:
Yes, I had thought about this. But the issue is that the methods used in CAPI create a [custom](https://github.com/kubernetes-sigs/cluster-api/blob/0f2f85c4f2623f57f59a64888c66d1782f415d3f/test/e2e/cluster_upgrade.go#L113) namespace of their own while we need to pass the namespace in [command](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/pull/639/files#diff-91bfcaef6c403f73cf5d595e5f61366278e68e2f58fd1e728a5e00bbf26efaf4R79) args in BYO host runner. Due to this, the namespace used in docker container to run agent will be different from the resources created in upgrade test case.

Reply:
Can we pass this custom namespace to byohost runner?

Reply:
We cannot pass the custom namespace to byohost runner since custom namespace is [randomly created](https://github.com/kubernetes-sigs/cluster-api/blob/0f2f85c4f2623f57f59a64888c66d1782f415d3f/test/e2e/cluster_upgrade.go#L113) at runtime inside methods of CAPI project and currently we do not have a way to control that namespace.

Reply:
@Madhur97 This is still in the `BeforeEach` right? What am I missing?

Reply:
Discussed offline. Makes sense to keep the tests separate